mitiq/interface/mitiq_qibo/conversions.py

```diff
@@ -0,0 +1,529 @@
+            raise UnsupportedQiboCircuitError(
+                f"{gate.name} is not supported by OpenQASM."
+            )
+
+    # Decompose the circuit in known Cirq gates
+    qasm = decompose_qibo_circuit(qibo_circuit).to_qasm()
+
+    return cirq_from_qasm(qasm)
+
+
+def to_qibo(circuit: Circuit) -> QiboCircuit:
+    """Returns a QiboCircuit equivalent to the input Cirq circuit.
+
+    Args:
+        circuit: Cirq circuit to convert to a QiboCircuit.
+
+    Returns:
+        QiboCircuit object equivalent to the input Mitiq circuit.
+    """
+    
+    try:  
+        qasm = cirq_to_qasm(circuit)
+        nqubits, gate_list = _parse_qasm_modified(qasm)
+        
+    except ValueError: 
+        decomp_circuit = Circuit(decompose(circuit)) 
+        qasm = cirq_to_qasm(decomp_circuit)
+        nqubits, gate_list = _parse_qasm_modified(qasm)
+    
+    if nqubits < 1:
+        raise UnsupportedCirqCircuitError(
+            f"The number of qubits must be at least 1 but is {nqubits}."
+        )
+
+    qibo_circuit = QiboCircuit(nqubits)
+    for gate_name, qubits, params in gate_list:
+        gate = getattr(gates, gate_name)
+        if gate_name == "M":
+            qibo_circuit.add(gate(*qubits, register_name=params))
+        elif params is None:
+            qibo_circuit.add(gate(*qubits))
+        else:
+            # assume parametrized gate
+            qibo_circuit.add(gate(*qubits, *params))
+    
+    return qibo_circuit
+
+
+def read_args(args: str) ->  Generator[Tuple[str, int], None, None]:
+        _args = iter(re.split(r"[\[\],]", args))
+        for name in _args:
+            if name:
+                index = next(_args)
+                if not index.isdigit():
+                    raise_error(ValueError, "Invalid QASM qubit arguments:", args)
+                yield name, int(index)
+
+
+def _parse_qasm_modified(qasm_code: str) -> Tuple[int, List[Tuple[str, List[int], Union[List[float], None]]]]:
+    """Extracts circuit information from QASM script.
+
+    Helper method for ``from_qasm``.
+
+    Args:
+        qasm_code: String with the QASM code to parse.
+
+    Returns:
+        nqubits: The total number of qubits in the circuit.
+        gate_list: List that specifies the gates of the circuit.
+            Contains tuples of the form
+            (Qibo gate name, qubit IDs, optional additional parameter).
+            The additional parameter is the ``register_name`` for
+            measurement gates or ``theta`` for parametrized gates.
+    """
+     
+    lines = (line for line in "".join(
+        line for line in qasm_code.split("\n") if line and line[:2] != "//" and line[1:3] != "//"
+    ).split(";") if line and "//" not in line)
+
+    if next(lines) != "OPENQASM 2.0":
+        raise_error(ValueError, "QASM code should start with 'OPENQASM 2.0'.")
+
+    qubits: Dict[Tuple[str, int], int] = {}  
+    cregs_size = {}  
+    registers: Dict[str, Optional[Dict[int, int]]] = {}
+    gate_list = []
```

Comment:
Not sure if this is the best solution, but since `mypy` is asking for type hints on the others, we could resolve another error with this type hint. @natestemen what do you think about this?
```suggestion
    gate_list: List[Tuple[str, List[int], Optional[List[float]]]] = []
```

Reply:
Yes! This solved some errors with the correct type indicator for gate_list: 
`gate_list: List[Tuple[str, List[int], Union[List[float], str, None]]] = []`
Without indicating the type mypy assumes the type of the list third argument after it's first assignment in the code wich might be either List[float],str or None, when trying to assign another type later on in the code mypy raised an error. Pre-assigning the type of gate_list solves this error.